disperser/batcher/encoded_blob_store.go

```diff
@@ -132,23 +139,28 @@ func (e *encodedBlobStore) DeleteEncodingResult(blobKey disperser.BlobKey, quoru
 	e.encodedResultSize -= getChunksSize(encodedResult)
 }
 
-// GetNewAndDeleteStaleEncodingResults returns all the fresh encoded results and deletes all the stale results
+// GetNewAndDeleteStaleEncodingResults returns all the fresh encoded results that are pending dispersal, and deletes all the stale results that are older than the given block number
 func (e *encodedBlobStore) GetNewAndDeleteStaleEncodingResults(blockNumber uint) []*EncodingResult {
 	e.mu.Lock()
 	defer e.mu.Unlock()
 	fetched := make([]*EncodingResult, 0)
 	staleCount := 0
+	pendingConfirmation := 0
 	for k, encodedResult := range e.encoded {
-		if encodedResult.ReferenceBlockNumber < blockNumber {
+		if encodedResult.Status == PendingConfirmation {
+			pendingConfirmation++
+		} else if encodedResult.ReferenceBlockNumber == blockNumber {
+			fetched = append(fetched, encodedResult)
+		} else if encodedResult.ReferenceBlockNumber < blockNumber {
 			// this is safe: https://go.dev/doc/effective_go#for
 			delete(e.encoded, k)
 			staleCount++
 			e.encodedResultSize -= getChunksSize(encodedResult)
 		} else {
```

Comment:
Curious about whether this case could actually happen? 

Possibilities: 
1. I guess since confirmation is happening in another thread, and we don't have locks (e.g. batcher.go, L288), it could be possible that the blob is marked as confirmed in the confirmation thread but not yet removed from the store. But not sure we need to handle that case here. 
2. A blob was encoded but somehow the confirmation transaction never took place. I guess this could probaby happen somehow.